packages/lesswrong/components/jargon/GlossaryEditForm.tsx

```diff
@@ -375,85 +464,95 @@ export const GlossaryEditForm = ({ classes, document, showTitle = true }: {
+      <div className={classNames(classes.headerButton, sortedApprovedTerms.length === 0 && classes.disabled)} 
+        onClick={() => handleSetApproveAll(false)}>
+        DISABLE ALL{ sortedApprovedTerms.length > 0 ?  ` (${sortedApprovedTerms.length})` : '' }
+      </div>
+    </LWTooltip>
+    <LWTooltip title={<div><p>Hide all terms that aren't currently enabled</p><p>(you can unhide them later)</p></div>}>
+      <div className={classNames(classes.headerButton, sortedUnapprovedTerms.length === 0 && classes.disabled)} onClick={handleDeleteUnused}>HIDE DISABLED TERMS</div>
+    </LWTooltip>
+    <LWTooltip title="Unhide all hidden terms">
+      <div className={classNames(classes.headerButton, deletedTerms.length === 0 && classes.disabled)} onClick={handleUnhideAll}>
+        UNHIDE ALL{ deletedTerms.length > 0 ? ` (${deletedTerms.length})` : '' }
+      </div>
+    </LWTooltip>
+  </div>
+
+  let rowCount = nonDeletedTerms.length
+  if (showDeletedTerms) { 
+    rowCount += deletedTerms.length
+  } else {
+    rowCount += deletedTerms.length > 0 ? 1 : 0
+  }
+
   const footer = <div className={classNames(classes.buttonRow, formCollapsed && classes.formCollapsed)}>
-    <Button onClick={addNewJargonTerms} className={classNames(classes.generateButton, mutationLoading && classes.disabled)}>Generate new terms</Button>
     <div className={classes.headerButtons}>
-      <LWTooltip title="Make changes to the jargon generator LLM prompt">
-        <div className={classes.headerButton} onClick={() => setEditingPrompt(!editingPrompt)}>{editingPrompt ? "SAVE PROMPT" : "EDIT PROMPT"}</div>
+    <LWTooltip title={showNewJargonTermForm ? "Cancel adding a new term" : "Manually add a new term to the glossary"}>
+      <div className={classNames(classes.newTermButton, showNewJargonTermForm && classes.newTermButtonCancel)} onClick={() => setShowNewJargonTermForm(!showNewJargonTermForm)}>
+        +
+      </div>
+      </LWTooltip>
+      <LWTooltip title={<div>Send post to {JARGON_LLM_MODEL_TITLE},<br/>to generate new glossary terms/explanations</div>}>
+        <Button onClick={autogenerateJargonTerms} className={classNames(classes.generateButton, (mutationLoading || !glossaryPrompt) && classes.disabled)}>Autogenerate Glossary</Button>
+      </LWTooltip>
+      <LWTooltip title="Edit the jargon generator LLM prompt">
+        <div className={classes.headerButton} onClick={() => setEditingPrompt(!editingPrompt)}>{editingPrompt ? "SAVE PROMPT" : <ForumIcon icon="Settings" className={classes.editPromptIcon} />}</div>
       </LWTooltip>
-    </div>
-    <div className={classes.button} onClick={() => setShowMoreTerms(!showMoreTerms)}>
-      {showMoreTerms 
-        ? <>SHOW LESS</>
-        : <>SHOW MORE</>
-      }
     </div>
     {mutationLoading && <Loading/>}
     {mutationLoading && <div>(Loading... warning, this will take 30-60 seconds.)</div>}
   </div>
 
+  const handleShowDeletedTerms = () => {
```

Comment:
move this up to be colocated with the state modification functions that it's wrapping around